src/lpf2hub.ts

```diff
@@ -608,5 +539,187 @@ export class LPF2Hub extends Hub {
+             */
+            this.emit("distance", port.id, distance);
+        }
+
+        if (color !== undefined && distance !== undefined) {
+            /**
+             * A combined color and distance event, emits when the sensor is activated.
+             * @event LPF2Hub#colorAndDistance
+             * @param {string} port
+             * @param {Color} color
+             * @param {number} distance Distance, in millimeters.
+             */
+            this.emit("colorAndDistance", port.id, color, distance);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.COUNT) {
+            /**
+             * Emits when a count mode distance sensor is activated.
+             * @event LPF2Hub#count
+             * @param {string} port
+             * @param {number} count
+             */
+            this.emit("count", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.REFLT) {
+            /**
+             * Emits when a reflect mode color sensor is activated.
+             * @event LPF2Hub#reflectivity
+             * @param {string} port
+             * @param {number} reflectivity
+             */
+            this.emit("reflectivity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.AMBI) {
+            /**
+             * Emits when a ambiant mode color sensor is activated.
+             * @event LPF2Hub#luminosity
+             * @param {string} port
+             * @param {number} luminosity
+             */
+            this.emit("luminosity", port.id, data[4]);
+        }
+
+        if (port.mode === Consts.ColorAndDistanceInputModes.RGB_I) {
+            /**
+             * Emits when a RGB mode color sensor is activated.
+             * @event LPF2Hub#rgb
+             * @param {string} port
+             * @param {number} r
+             * @param {number} g
+             * @param {number} b
+             */
+            this.emit("rgb", port.id, data[4], data[6], data[8]);
+        }
+    }
+
+    private _parseBoostTilt(port: Port, data: Buffer) {
+        const values: number[] = [];
+        let event = '';
+
+        switch (port.mode) {
+            case Consts.BoostTiltModes.ANGLE: {
+                values.push(data.readInt8(4));
+                values.push(data.readInt8(5));
+                event = 'angle';
+                break;
+            }
+
+            case Consts.BoostTiltModes.TILT: {
+                values.push(data[4]);
+                event = 'tilt';
+                break;
+            }
+
+            case Consts.BoostTiltModes.ORINT: {
+                values.push(data[4]);
+                event = 'orientation';
+                break;
+            }
+
+            case Consts.BoostTiltModes.IMPCT: {
+                values.push(data.readUInt32BE(4));
+                event = 'impact';
+                break;
+            }
+
+            case Consts.BoostTiltModes.ACCEL: {
```

Comment:
This change makes the API incompatible since previously this mode, which is the default mode for this sensor was reported as "tilt".

Reply:
I don't know what should be done as it breaks but on the other hand, it use information from the lego side which should be "the truth".

There is a lot of changes in this PR and it definitly could not be merge without a new major version.